src/iam.ts

```diff
@@ -350,11 +353,17 @@ export class IAM extends IAMBase {
       throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
     }
 
-    const { iss, sub, claimData } = (await this.decodeJWTToken({ token })) as {
+    const payload = (await this.decodeJWTToken({ token })) as {
       iss: string;
       sub: string;
       claimData: ClaimData;
     };
+    const { iss, claimData } = payload;
+    let sub = payload.sub;
+    if (!sub || sub.length === 0) {
```

Comment:
@JGiter I believe that previously, claims had the role namespace as the `sub`, so this is what the code should detect and should replace the `sub` to be `this.did` in this case as well.

example of old incorrect claim:
```
{
  "did": "prequalified.roles.flexmarket.apps.elia.iam.ewc",
  "signer": "did:ethr:0x829b91Fa3e91EA4448365ADA58C7Bad1Ff142866",
  "claimData": {
    "fields": [],
    "claimType": "prequalified.roles.flexmarket.apps.elia.iam.ewc"
  },
  "sub": "prequalified.roles.flexmarket.apps.elia.iam.ewc",
  "iat": 1614851452761,
  "iss": "did:ethr:0x829b91Fa3e91EA4448365ADA58C7Bad1Ff142866"
}
```

Reply:
Added checking. However, it is strange why the subject was the role